### PR TITLE
Sized floats and ints

### PR DESCRIPTION
**Pull request by nahiim**
## Description
Use sized floats and ints in entire HLSL codebase

### CODE REVIEW

**Comment by devshgraphicsprogramming** [comment]
can you please BxDF stuff and the rest separate

**Comment by nahiim** [comment]
> can you please BxDF stuff and the rest separate

I should have changes per directory in different commits?

File: include/nbl/builtin/hlsl/algorithm.hlsl [hunk 1]
```diff
@@ -90,14 +90,14 @@ struct lower_to_upper_comparator_transform_t
...
-uint lower_bound(inout Accessor accessor, const uint begin, const uint end, const typename Accessor::value_type value, const Comparator comp)
+uint32_t lower_bound(inout Accessor accessor, const uint32_t begin, const uint32_t end, const typename Accessor::value_type value, const Comparator comp)
 {
     impl::bound_t<Accessor,Comparator> implementation = impl::bound_t<Accessor,Comparator>::setup(begin,end,value,comp);
     return implementation(accessor);
 }
 
 template<class Accessor, class Comparator>
-uint upper_bound(inout Accessor accessor, const uint begin, const uint end, const typename Accessor::value_type value, const Comparator comp)
+uint32_t upper_bound(inout Accessor accessor, const uint32_t begin, const uint32_t end, const typename Accessor::value_type value, const Comparator comp)
```
Comment (devshgraphicsprogramming) [inline include/nbl/builtin/hlsl/algorithm.hlsl:100]: `inout` needs to get replaced by` NBL_ARG_REF(T)`

Comment (nahiim) [reply include/nbl/builtin/hlsl/algorithm.hlsl:100]: I was gonna do that in a seperate PR.

**Comment by devshgraphicsprogramming** [comment]
> > can you please BxDF stuff and the rest separate
> 
> I should have changes per directory in seperate commits?

separate PRs / branches

**Comment by devshgraphicsprogramming** [comment]
bwtw don't touch anything in the `bxdf` directory

**Comment by nahiim** [comment]
> bwtw don't touch anything in the `bxdf` directory

I already changed the ints and floats. And the "out" variables. Should I revert?

**Comment by devshgraphicsprogramming** [comment]
> > bwtw don't touch anything in the `bxdf` directory
> 
> I already changed the ints and floats. And the "out" variables. Should I revert?

don't revert, just abandon the branch and copy over what you need to some other branch

**Pull request closed without merging**